src/cljs/owlet_ui/components/branch.cljs

```diff
@@ -4,6 +4,20 @@
             [reagent.core :as reagent :refer [atom]]
             [camel-snake-kebab.core :refer [->kebab-case]]))
 
+(def hover-image (reagent/atom ""))
+
+(defn get-random [images]
+  (get images (rand-int (count images))))
+
+(defn set-hover [images]
+  (let [image (get-random images)]
+    (if (> (count images) 1)
+      ; prevent the same image twice in a row
+      (if (== image @hover-image)
+        (set-hover images)
```

Comment:
Two things:
1. the reason for `recur` in clojure has more to do with Java's lack of proper tail call optimizations:
> While Clojure is a Lisp-based language, it also runs on top of a host virtual machine (usually the JVM), and currently the JVM does not support tail recursion.
[source](http://www.howardism.org/Technical/Clojure/tail-recursion.html)

I wonder if applies to ClojureScript at all since the host is JavaScript . It might not matter as much [in the future](http://www.2ality.com/2015/06/tail-call-optimization.html).

2. There is nothing wrong with this code per say. It might be more performant to use `recur` but that depends on the underlying implementation. When tail recursion is present in the language and the host doesn't get in the way, having a recursive function like this is perfectly idiomatic. In Erlang/Elixir for example this is not even an issue. In fact it is the only way to loop (this doesn't mean that Lisp in general are bad at recursion, in fact recursion was first introduced by Lisp! Scheme has tail recursion for example).